app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -154,9 +155,15 @@ public void updateIfNeeded() {
         }
 
         /* Section - GUI */
-        // Enforce TLS.
         Element gui = getGuiElement();
-        changed = setConfigElement(gui, "tls", "true") || changed;
+
+        // Platform-specific: Force REST API and Web UI access to use TLS 1.2 or not.
+        Boolean forceHttps = Constants.osSupportsTLS12();
+        if (!gui.hasAttribute("tls") ||
+                Boolean.parseBoolean(gui.getAttribute("tls")) != forceHttps) {
+            gui.setAttribute("tls", forceHttps ? "true" : "false");
```

Comment:
Just ` gui.setAttribute("tls", forceHttps)`

Reply:
Would work, but we miss detecting the change then and according to "changed" we need to save the config to disk.

Reply:
I don't mean the entire block, just that `forceHttps ? "true" : "false"`. However I didn't notice the quotes around true/false: Seems like `setAttribute` takes a string not a bool. That seems quite bad to me, but unrelated to this PR.

Reply:
Yes, In wonder if we should have constants for that as it hasn't to match Java but what SyncthingNative expects when reading the config.